<!DOCTYPE html>
<!-- This site was created with Hugo Blox. https://hugoblox.com -->
<!-- Last Published: October 12, 2025 --><html lang="en-us" >


<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  
  
  
    <meta name="generator" content="Hugo Blox Builder 5.9.7" />
  

  
  












  
  










  







  
  
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
  

  
  
  
    
      
      <link rel="preload" as="style" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap">
      <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap" media="print" onload="this.media='all'">
    
  

  
  

  
  
    
    <script src="/js/mathjax-config.js"></script>
  

  

  <link rel="stylesheet" href="/css/vendor-bundle.min.css" media="print" onload="this.media='all'">

  
  
  
    
    
      <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jpswalsh/academicons@1.9.4/css/academicons.min.css" integrity="sha512-IW0nhlW5MgNydsXJO40En2EoCkTTjZhI3yuODrZIc8cQ4h1XcF53PsqDHa09NqnkXuIe0Oiyyj171BqZFwISBw==" crossorigin="anonymous" media="print" onload="this.media='all'">
    

    
    
    
    
      
      
    
    
    

    
    
    
      <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/leaflet@1.7.1/dist/leaflet.min.css" integrity="" crossorigin="anonymous" media="print" onload="this.media='all'">
    

    

    
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js" integrity="" crossorigin="anonymous" async></script>
      
    
      
      

      
      

      
    
      
      

      
      

      
    
  

  
  
  
  
  
  
  <link rel="stylesheet" href="/css/wowchemy.css" />

  
  
  

  
  
  
  
  
  
  
    
    
    <link rel="stylesheet" href="/css/libs/chroma/github-light.min.css" title="hl-light" media="print" onload="this.media='all'" >
    <link rel="stylesheet" href="/css/libs/chroma/dracula.min.css" title="hl-dark" media="print" onload="this.media='all'" disabled>
  

  
  
























  
  
  






  <meta name="author" content="Daniel Fridljand" />





  

<meta name="description" content="The Hierarchical Dirichlet Process (HDP) is a popular and elegant model that has gained traction in various fields, including cancer genomics. While researching methods for mutational signature estimation, I noticed that several studies employed HDPs for this purpose. However, many of these studies were published in biological journals and often lacked detailed explanations of how HDPs can be effectively applied to mutational signature estimation." />



<link rel="alternate" hreflang="en-us" href="http://localhost:1313/post/mutational-signature-with-hierarchical-dirichlet-process/" />
<link rel="canonical" href="http://localhost:1313/post/mutational-signature-with-hierarchical-dirichlet-process/" />



  <link rel="manifest" href="/manifest.webmanifest" />



<link rel="icon" type="image/png" href="/media/icon_hu_6698497dad6a3c8a.png" />
<link rel="apple-touch-icon" type="image/png" href="/media/icon_hu_f7c1592650856cd8.png" />

<meta name="theme-color" content="#1565c0" />










  






<meta property="twitter:card" content="summary_large_image" />

  <meta property="twitter:site" content="@GetResearchDev" />
  <meta property="twitter:creator" content="@GetResearchDev" />
<meta property="twitter:image" content="http://localhost:1313/post/mutational-signature-with-hierarchical-dirichlet-process/featured.png" />



  

<meta property="og:type" content="article" />
<meta property="og:site_name" content="Daniel Fridljand" />
<meta property="og:url" content="http://localhost:1313/post/mutational-signature-with-hierarchical-dirichlet-process/" />
<meta property="og:title" content="Mutational signature estimation with Hierarchical Dirichlet Process | Daniel Fridljand" />
<meta property="og:description" content="The Hierarchical Dirichlet Process (HDP) is a popular and elegant model that has gained traction in various fields, including cancer genomics. While researching methods for mutational signature estimation, I noticed that several studies employed HDPs for this purpose. However, many of these studies were published in biological journals and often lacked detailed explanations of how HDPs can be effectively applied to mutational signature estimation." /><meta property="og:image" content="http://localhost:1313/post/mutational-signature-with-hierarchical-dirichlet-process/featured.png" /><meta property="og:locale" content="en-us" />

  
    <meta
      property="article:published_time"
      content="2024-09-15T00:00:00&#43;00:00"
    />
  
  
    <meta property="article:modified_time" content="2024-09-15T00:00:00&#43;00:00">
  






    






  




<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "http://localhost:1313/post/mutational-signature-with-hierarchical-dirichlet-process/"
  },
  "headline": "Mutational signature estimation with Hierarchical Dirichlet Process",
  
  "image": [
    "http://localhost:1313/post/mutational-signature-with-hierarchical-dirichlet-process/featured.png"
  ],
  
  "datePublished": "2024-09-15T00:00:00Z",
  "dateModified": "2024-09-15T00:00:00Z",
  
  "author": {
    "@type": "Person",
    "name": "Daniel Fridljand"
  },
  
  "publisher": {
    "@type": "Organization",
    "name": "Daniel Fridljand",
    "logo": {
      "@type": "ImageObject",
      "url": "http://localhost:1313/media/icon_hu_c928bdf40dddf8a.png"
    }
  },
  "description": "\u003cp\u003eThe Hierarchical Dirichlet Process (HDP) is a popular and elegant model that has gained traction in various fields, including cancer genomics. While researching methods for mutational signature estimation, I noticed that several studies employed HDPs for this purpose. However, many of these studies were published in biological journals and often lacked detailed explanations of how HDPs can be effectively applied to mutational signature estimation.\u003c/p\u003e"
}
</script>

  

  




  
  
  

  
  

  


  
  <title>Mutational signature estimation with Hierarchical Dirichlet Process | Daniel Fridljand</title>

  
  
  
  











</head>


<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" class="page-wrapper   " data-wc-page-id="cddf9e95d42f1f866f0d4b590fd169df" >

  
  
  
  
  
  
  
  
  
  <script src="/js/wowchemy-init.min.js"></script>

  


<aside class="search-modal" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#" aria-label="Close"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search" class="form-control"
        aria-label="Search...">
        
      </div>

      
      

      

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>



  <div class="page-header header--fixed">
  
  
  
  
  












<header>
  <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
    <div class="container-xl">

      
      <div class="d-none d-lg-inline-flex">
        <a class="navbar-brand" href="/">Daniel Fridljand</a>
      </div>
      

      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar-content" aria-controls="navbar-content" aria-expanded="false" aria-label="Toggle navigation">
      <span><i class="fas fa-bars"></i></span>
      </button>
      

      
      <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
        <a class="navbar-brand" href="/">Daniel Fridljand</a>
      </div>
      

      
      
      <div class="navbar-collapse main-menu-item collapse justify-content-start" id="navbar-content">

        
        <ul class="navbar-nav d-md-inline-flex">
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
              
              
              
                
              
              
            
          

          <li class="nav-item">
            <a class="nav-link " href="/#about"><span>Home</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
              
              
              
                
              
              
            
          

          <li class="nav-item">
            <a class="nav-link " href="/#posts"><span>Posts</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
              
              
              
                
              
              
            
          

          <li class="nav-item">
            <a class="nav-link " href="/#experience"><span>Experience</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
              
              
              
                
              
              
            
          

          <li class="nav-item">
            <a class="nav-link " href="/#publications"><span>Publications</span></a>
          </li>

          
          

        

          
        </ul>
      </div>

      <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">

        
        
          
        

        
        
        
        <li class="nav-item">
          <a class="nav-link js-search" href="#" aria-label="Search"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        
        
        
        <li class="nav-item dropdown theme-dropdown">
          <a href="#" class="nav-link" data-toggle="dropdown" aria-haspopup="true" aria-label="Display preferences">
            <i class="fas fa-moon" aria-hidden="true"></i>
          </a>
          <div class="dropdown-menu">
            <a href="#" class="dropdown-item js-set-theme-light">
              <span>Light</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-dark">
              <span>Dark</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-auto">
              <span>Automatic</span>
            </a>
          </div>
        </li>
        

        
        

      </ul>

    </div>
  </nav>
</header>


  </div>

  <div class="page-body">
    
    
    

    <article class="article">

  






















  
  



<div class="article-container pt-3">
  <h1>Mutational signature estimation with Hierarchical Dirichlet Process</h1>

  

  


<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    Sep 15, 2024
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    12 min read
  </span>
  

  
  
  
  

  
  

</div>

  





</div>


<div class="article-header container featured-image-wrapper mt-4 mb-4" style="max-width: 663px; max-height: 409px;">
  <div style="position: relative">
    <img src="/post/mutational-signature-with-hierarchical-dirichlet-process/featured_hu_6c8c50c733226c55.webp" width="663" height="409" alt="" class="featured-image">
    <span class="article-header-caption">Visualised mutational signature</span>
  </div>
</div>



  <div class="article-container">

    <div class="article-style">
      <p>The Hierarchical Dirichlet Process (HDP) is a popular and elegant model that has gained traction in various fields, including cancer genomics. While researching methods for mutational signature estimation, I noticed that several studies employed HDPs for this purpose. However, many of these studies were published in biological journals and often lacked detailed explanations of how HDPs can be effectively applied to mutational signature estimation.</p>
<p>Nicola Roberts&rsquo; <a href="https://doi.org/10.17863/CAM.22674" target="_blank" rel="noopener">dissertation</a> includes some descriptions of the mathematical model in the appendix, but even these are not fully comprehensive. In this blog post, I aim to bridge this gap by introducing the problem of mutational signature estimation using HDPs in a mathematically rigorous manner. I will begin by discussing a simpler mixture component model for mutational signatures, then explore the Dirichlet Process model, and finally delve into the Hierarchical Dirichlet Process model.</p>
<h2 id="introduction">Introduction</h2>
<p>A key characteristic of cancer is the elevated mutation rate in somatic cells. Recognizing the patterns of somatic mutations in cancer is essential for understanding the underlying mechanisms that drive the disease (Fischer et al., 2013). This knowledge is crucial for advancing cancer treatment and prevention.</p>
<p>One type of somatic mutation is the single nucleotide variant (SNV), also referred to as a single base substitution. This involves the replacement of one base pair in the genome with another. There are six possible base pair mutations: C:G&gt;A:T, C:G&gt;G:C, C:G&gt;T:A, T:A&gt;A:T, T:A&gt;C:G, and T:A&gt;G:C. When considering the context of the immediate 5&rsquo; and 3&rsquo; neighboring bases, the total number of mutations increases to 96, calculated as $6 \times 4 \times 4 = 96$. These are known as trinucleotide mutation channels. Examples of such mutations include ACA:TGT&gt;AAA:TTT and ACT:TGA&gt;AGT:TCA. For simplicity, we will abbreviate these mutations as ACA&gt;AAA and ACT&gt;AGT respectively.</p>
<h2 id="mutational-signature">Mutational Signature</h2>
<p>A mutational signature $\theta$ with respect to trinucleotide mutation channels is a discrete probability distribution over the 96 mutational channels i.e. $\theta\in \Delta^{96}$ where</p>



$$
\Delta^{n} = \left\{ (x_1, x_2, \ldots, x_{n}) \in \mathbb{R}^{n} \mid x_i \geq 0 \text{ for all } i \text{ and } \sum_{i=1}^{n} x_i = 1 \right\}.
$$

<p>A mutational signature $\theta$ can be visualised as in the following.</p>
<figure>
<img src="media/a9711185bb2959bf64bf6004e591ab8de652fedf.png"
 alt="500" />
</figure>
<p>The framework for studying somatic mutations through mutational signatures was introduced in a landmark study by Alexandrov et al. (2013), where over 7,000 bulk-sequenced cancer samples were analyzed. Conceptually, a mutational signature represents a biological process acting on the genome, leaving a distinct imprint captured in a probability vector, $\theta$. A comprehensive list of mutational signatures and their proposed underlying biological processes is available through the <a href="https://cancer.sanger.ac.uk/signatures/sbs/" target="_blank" rel="noopener">COSMIC project</a>. Some of the mutational signatures and their associated probability vectors $\theta$ in the COSMIC project are consistently identifiable across most examined cohorts and signature identification algorithms <a href="https://pubmed.ncbi.nlm.nih.gov/34316057/" target="_blank" rel="noopener">(cite)</a>.</p>
<p>To estimate mutational signatures and their activities, <a href="https://pubmed.ncbi.nlm.nih.gov/23318258/" target="_blank" rel="noopener">Alexandrov et al.</a> employed Non-Negative Matrix Factorization (NMF). NMF and its variations remain the most widely used methods for estimating mutational signatures and mutational signature activities.</p>
<h2 id="dirichlet-process-dp">Dirichlet Process (DP)</h2>
<p>A key mathematical tool for our study is the Dirichlet Process (DP). Formally, the Dirichlet process, denoted as $\mathrm{DP}(\alpha, H)$, is a distribution over probability measures. When we sample $G \sim \mathrm{DP}(\alpha, H)$, we obtain a probability measure $G$. The Dirichlet Process has two parameters: the concentration parameter $\alpha&gt;0$, and the base measure $H$. The support of $G$ is a subset of the support of the base measure $H$.</p>
<p>To sample from the Dirichlet process, we begin by drawing an infinite sequence of i.i.d. samples $\tilde{\theta}_1, \tilde{\theta}_2, \ldots \sim H$. Then, we draw a vector $\boldsymbol{e} = \{e_1, e_2, \ldots\} \sim \operatorname{Stick}(\alpha)$, where $\operatorname{Stick}(\alpha)$ refers to the stick-breaking process. Intuitively, the stick-breaking process works by repeatedly breaking off and discarding a random fraction of a &ldquo;stick&rdquo; that is initially of length $1$. Each broken off part of the stick has a length $e_k$. The stick-breaking process generates a sequence $\boldsymbol{e} \in \Delta^\infty$, where the total length of all the broken pieces is 1.</p>
<p>Finally, we construct the random measure $G$ as:



$$G = \sum_{k=1}^{\infty} e_k \delta_{\tilde{\theta}_k}$$

where $\delta_{\tilde{\theta}_k}$ is a Dirac delta function centered at $\tilde{\theta}_k$. The resulting measure $G$ is discrete, with countable support given by $\{\tilde{\theta}_1, \tilde{\theta}_2, \ldots\}$.</p>
<h3 id="mathematical-properties-of-dirichlet-process">Mathematical properties of Dirichlet Process</h3>
<p>The base measure $H$ serves as the mean of the random measure $G$. Specifically, for any measurable set $A$, the expectation of $G(A)$ under the Dirichlet process is given by [@millerDirichletProcessModels]:</p>



$$
\mathrm{E}_{G \sim \mathrm{DP}(\alpha, H)}[G(A)]=H(A)
$$

<p>As $\alpha \rightarrow \infty$, the distribution of $G$ converges to $H$ in the weak topology, meaning that larger values of $\alpha$ make $G$ more closely resemble the base measure $H$.</p>
<p>In summary, $G$ is centered around the prior distribution $H$, with the concentration parameter $\alpha$ controlling how tightly $G$ clusters around $H$. A higher $\alpha$ reduces variability, causing $G$ to approximate $H$ more closely.</p>
<h3 id="references">References</h3>
<details>
<summary>
<p>Recommended material for Dirichlet Process</p>
</summary>
<ul>
<li><a href="https://en.wikipedia.org/wiki/Dirichlet_process" target="_blank" rel="noopener">https://en.wikipedia.org/wiki/Dirichlet_process</a></li>
<li><a href="https://www.geeksforgeeks.org/dirichlet-process-mixture-models-dpmms/" target="_blank" rel="noopener">https://www.geeksforgeeks.org/dirichlet-process-mixture-models-dpmms/</a></li>
<li><a href="https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-d259c237a840" target="_blank" rel="noopener">https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-d259c237a840</a></li>
<li><a href="https://pawel-czyz.github.io/posts/dirichlet-process.html" target="_blank" rel="noopener">https://pawel-czyz.github.io/posts/dirichlet-process.html</a></li>
</ul>
</details>
<details>
<summary>
<p>NoteBook Impelementation of the Dirichlet Process</p>
</summary>
<ul>
<li><a href="https://datamicroscopes.github.io/ncluster.html" target="_blank" rel="noopener">https://datamicroscopes.github.io/ncluster.html</a></li>
<li><a href="https://github.com/tdhopper/notes-on-dirichlet-processes/tree/master" target="_blank" rel="noopener">https://github.com/tdhopper/notes-on-dirichlet-processes/tree/master</a></li>
<li><a href="https://github.com/tdhopper/notes-on-dirichlet-processes/blob/master/pages/2015-07-28-dirichlet-distribution-dirichlet-process.ipynb" target="_blank" rel="noopener">https://github.com/tdhopper/notes-on-dirichlet-processes/blob/master/pages/2015-07-28-dirichlet-distribution-dirichlet-process.ipynb</a></li>
</ul>
</details>
<h2 id="modelling-mutational-signature-with-dp">Modelling mutational signature with DP</h2>
<p>Dirichlet processes have been utilized in previous studies to model trinucleotide mutations associated with mutational signatures <a href="https://doi.org/10.17863/CAM.22674" target="_blank" rel="noopener">(cite)</a> and <a href="https://www.nature.com/articles/s41586-019-1913-9" target="_blank" rel="noopener">(cite)</a>. Although Appendix B (p. 234) and Chapter 4 (p. 132) in <a href="https://doi.org/10.17863/CAM.22674" target="_blank" rel="noopener">(cite)</a> provide a mathematical overview of the model, they lack sufficient detail and are not fully comprehensive. This section aims to address that gap by deriving the Dirichlet Process model for trinucleotide mutations in a thorough and accessible manner. We will begin by introducing a more intuitive approach to modeling trinucleotide mutations using a mixture model. To illustrate, consider the trinucleotide mutations within a single sample, which can be encoded as $x \in \{1, \ldots, 96\}$, representing the 96 possible trinucleotide mutation types.</p>
<h4 id="step-1-known-number-of-mutational-signatures-k">Step 1: Known number of mutational signatures $K$</h4>
<p>Let&rsquo;s begin by assuming we know the number of active mutational signatures, denoted as $K$. Each signature is represented by a discrete probability distribution, $\tilde{\theta}_k \in \Delta^{96}$, where $k=1, \ldots, K$. Since we assume that we have no prior knowledge of the specific mutational signatures, we model these distributions as:



$$
\begin{aligned}
\tilde{\theta}_k  & \sim \operatorname{Dir}\left(\frac{1}{96} \cdot \mathbf{1}_{96}\right) 
\text{ for } k=1,\ldots, K\\
\end{aligned},
$$

where $\operatorname{Dir}$ denotes the Dirichlet distribution, serving as a symmetric prior. Similarly, we assume no prior information about the activity of these mutational signatures. We model the activity of the $K$ mutational signatures $\boldsymbol{e} \in \Delta^K$ in a sample as:



$$
\begin{aligned}
\boldsymbol{e} \mid \alpha,K & \sim \operatorname{Dir}\left(\frac{\alpha}{K} \cdot \mathbf{1}_K\right). 
\end{aligned}
$$

where $\alpha$ is a hyperparameter controlling the concentration of the distribution. Next, let $z_i=k$ represent the event that the $i$-th mutation was generated by the $k$-th mutational signature. If we observe a total of $M$ trinucleotide mutations in our sample, we draw each $z_i\in \{{1,\ldots, K}\}$ according to the mutational signature activity $\boldsymbol{e}$, as follows:</p>



$$
\begin{aligned}
 z_i \mid \boldsymbol{e} & \sim \operatorname{Categorical} \left(\boldsymbol{e}\right)  \text{ for } i=1,\ldots, M,
\end{aligned}
$$

<p>indicating that the probability $\mathrm{P}\left(z_i=k\right)=e_k$, where $e_k$ is the activity level of signature $k$. Finally, the observed trinucleotide mutation $x_i \in\{1, \ldots, 96\}$ is drawn from a categorical distribution based on the corresponding mutational signature $z_i=k$ and its distribution $\tilde{\theta}_k$ :</p>
<p>


$$
\begin{aligned}
\{{1,\ldots, 96}\} \ni x_i \mid z_i=k, \tilde{\theta}_k & \sim \operatorname{Categorical} \left(\tilde{\theta}_{k}\right)  \text{ for } i=1,\ldots, M
\end{aligned}
$$

This completes our model, which describes how the mutational signatures $\left\{\tilde{\theta}_1, \ldots, \tilde{\theta}_K\right\}$ generate the observed trinucleotide mutations $\left\{x_1, \ldots, x_M\right\}$</p>
<h5 id="graphical-model-for-known-number-of-components-k">Graphical model for known number of components $K$</h5>
<p>We can visualize the process that generates trinucleotide mutations in a sample using a graphical model:</p>
<figure>
<img src="media/7d5261339cd5d516f8153605fe36589a23255d8d.png"
 alt="300" />
</figure>
<p>Here the grey background color indicates that $x_i$ is an observed random variable.</p>
<h4 id="step-2-equivalent-model-as-mixture">Step 2: Equivalent model as mixture</h4>
<p>In the next step, we aim to integrate out the indicator variable $z_i=k$ in our model, seeking a different representation of the same model. As before, we draw the mutational signature activities as follows:



$$
\begin{aligned}
\boldsymbol{e} \mid \alpha,K & \sim \operatorname{Dir}\left(\frac{\alpha}{K} \cdot \mathbf{1}_K\right). 
\end{aligned}
$$

Let $H=\operatorname{Dir}\left(\frac{1}{96} \cdot \mathbf{1}_{96}\right)$. Similar to the last step, we generate the mutational signatures as



$$
\begin{aligned}
\tilde{\theta}_k & \sim H \text{ for } k=1,\ldots, K.
\end{aligned}
$$
</p>
<p>Now, instead of assuming that each observation $i=1, \ldots, M$ is first assigned a mutational signature $z_i \in\{1, \ldots, K\}$ and then drawn from the corresponding signature $\tilde{\theta}_{z_i}$ we directly draw $\theta_i$ from $\left\{\tilde{\theta}_1, \ldots, \tilde{\theta}_K\right\}$, with probabilities $\left\{e_1, \ldots, e_K\right\}$ for each observation:</p>
<p>


$$
\begin{aligned}
\theta_i \mid \boldsymbol{e}, \tilde{\theta}_1, \ldots \tilde{\theta}_K & \sim G=\sum_{k=1}^K e_k \delta_{\tilde{\theta}_k}\left(\theta_i\right) \text{ for } i=1,\ldots, M.
\end{aligned}
$$

Here, $G$ represents a finite mixture distribution with support on the acting mutational signatures. Finally, we draw the trinucleotide mutation as follows:



$$
\begin{aligned}
x_i \mid \theta_i & \sim \operatorname{Categorical} \left(\theta_i\right) \text{ for } i=1,\ldots, M. 
\end{aligned}
$$
</p>
<h4 id="step-3-unknown-unbounded-number-of-components-k">Step 3: Unknown, unbounded number of components $K$</h4>
<p>Let



$$
\Delta^{\infty} = \left\{ (\theta_1, \theta_2, \ldots, ) \in \mathbb{R}^{\infty} \mid \theta_i \geq 0 \text{ for all } i \text{ and } \sum_{i=1}^{\infty} \theta_i = 1 \right\}.
$$
</p>
<p>We now extend the model from the previous step to allow for an unknown number of components $K$, with any $K \in \mathbb{N}$ being possible. As in the previous step, we generate the underlying mutational signatures $\tilde{\theta}_k\in \Delta^{96}$ from $H$ as follows:



$$
\begin{aligned}
\tilde{\theta}_k & \sim H \text{ for } k=1,2,\ldots 
\end{aligned}
$$

Previously, we sampled a finite set of $K$ mutational signatures $\tilde{\theta}_k$, but now we generate an infinite sequence of signatures. Accordingly, the mutational signature activities $\boldsymbol{e}$ now lie in $\Delta^{\infty}$. To model these activities, we use the Stick-breaking process, which defines a distribution over $\Delta^{\infty}$. Thus, we sample the mutational signature activities as:</p>
<p>


$$
\begin{aligned}
\boldsymbol{e} \mid \alpha &\sim \operatorname{Stick}(\alpha).
\end{aligned}
$$

Intuitively, the stick-breaking process works by repeatedly breaking off and discarding a random fraction of a &ldquo;stick&rdquo; that is initially of length $1$. Each broken off part of the stick has a length $e_k$. The stick-breaking process generates a sequence $\boldsymbol{e} \in \Delta^\infty$, where the total length of all the broken pieces is $1$. Consequently, the mixture distribution $G$, which previously had finite support, now has countably infinite support $\left\{\tilde{\theta}_1, \tilde{\theta}_2, \ldots\right\}$ and takes the form:



$$
\begin{aligned}
\mathcal{P}(\Delta^{96}) \ni G&=\sum_{k=1}^{\infty} e_k \delta_{\tilde{\theta}_k}\left(\theta_i\right) , 
\end{aligned}
$$

where $\mathcal{P}(\Delta^{96})$ denotes the space of probability distributions on $\Delta^{96}$. We draw $\theta_i$ from the support of $G$, which is $\left\{\tilde{\theta}_1, \tilde{\theta}_2, \ldots\right\}$, as follows:



$$
\begin{aligned}
\theta_i \mid  G & \sim G \text{ for } i=1,\ldots, M \\  
\end{aligned}
$$
</p>
<p>Finally, we observe the mutations as



$$
\begin{aligned}
\{{1,\ldots, 96}\} \ni x_i \mid \theta_i & \sim \operatorname{Categorical} \left(\theta_i\right) \text{ for } i=1,\ldots, M 
\end{aligned}
$$
</p>
<h4 id="step-4-short-hand-notation">Step 4: Short hand notation</h4>
<p>We summarize the first three lines in the last equation and say that $G$ was genereated by the Dirichlet Process $\mathrm{DP}$ with the parameters $\alpha$ and $H$. This way the above equations simplify to</p>



$$
\begin{aligned}
G &\sim \mathrm{DP}(\alpha, H)\\ 
\Delta^{96}\ni \theta_i \mid G &\sim G \text{ for } i=1,\ldots, M \\
\{{1,\ldots, 96}\} \ni x_i \mid \theta_i & \sim \operatorname{Categorical} \left(\theta_i\right) \text{ for } i=1,\ldots, M   
\end{aligned}
$$

<p>This concludes our derivation of how we can model trinucleotide mutations with a Dirichlet Process, as used in <a href="https://www.nature.com/articles/s41586-019-1913-9" target="_blank" rel="noopener">(cite)</a>. The Dirichlet Process approach presents an alternative approach to estimate mutational signature and signature activity as alternative to the classical approach based on Non-negative Matrix Factorisation used in <a href="https://www.nature.com/articles/nature12477" target="_blank" rel="noopener">(cite)</a>.</p>
<h5 id="graphical-model-of-dirichlet-process">Graphical Model of Dirichlet Process</h5>
<p>We can visualise the Dirichlet process as a graphical model:</p>
<figure>
<img src="media/cae3659c625ecead8e187392389b2cc3e01df00c.png"
 alt="ThegraphicalmodelofaDirichletProcess." />
<figcaption aria-hidden="true">ThegraphicalmodelofaDirichletProcess.</figcaption>
</figure>
<h2 id="hierarchical-dirichlet-process-hdp">Hierarchical Dirichlet Process (HDP)</h2>
<p>By taking composition of multiple Dirichlet Processes, we obtain a Hierarchical Dirichlet Process (HDP) <a href="https://www.tandfonline.com/doi/abs/10.1198/016214506000000302" target="_blank" rel="noopener">(cite)</a>. Recall that Dirichlet processes excel at modelling mixtures with an unknown number of components. HDPs are tailored for the situation when we have groups of mixture components and/or hierarchies between mixture components. For our model of trinucleotide mutations, we can use HDP to account for multiple samples. We expect from biology that there are general trends of mutational signature activity. However, we expect those trends to be more similar within a sample than between samples. In a mixture modelling approach, this would be captured by separate mixture models for each sample and information sharing between the weights of each sample. In the Dirichlet Process approach, this can be reflected by the HDP model. As before, we start with the prior distribution $H$. We draw a common base distribution $G_0$ from a Dirichlet Process as



$$
\begin{aligned}
G_0 &\sim \mathrm{DP}(\alpha_0, H).\\ 
\end{aligned}
$$

Let&rsquo;s assume we have $N$ samples. For each sample, we draw a seperate Dirichlet Process $G_j$ from the common base distribution $G_0$ as



$$
\begin{aligned}
G_j \mid G_0 &\sim \mathrm{DP}(\alpha_j, G_0) \text{ for } j = 1,2,\ldots N. 
\end{aligned}
$$

In other words, we have a Dirichlet Process $G_0$ at the top and other Dirichlet Processes $G_j$ at the lower hierarchies. As described in the section of the mathematical properties of the Dirichlet Process, each $G_j$ will vary around $G_0$. This allows the different samples $G_j$ to share information via $G_0$. Let&rsquo;s assume we observe $M_j$ mutations in the $j$-th sample. We draw the mutations as before:</p>



$$
\begin{aligned}
\theta_{j i} \mid G_j &\sim G_j \text{ for } i = 1,2,\ldots M_j , \\
x_{ji} \mid \theta_{j i} & \sim \operatorname{Categorical} \left(\theta_{j i}\right). 
\end{aligned}
$$

<p>The corresponding graphical model is:</p>
<figure>
<img src="media/7d5261339cd5d516f8153605fe36589a23255d8d.png"
 alt="500" />
</figure>
<h3 id="estimation-of-mutational-signature-and-mutational-signature-activity">Estimation of mutational signature and mutational signature activity</h3>
<p>Let&rsquo;s assume we have the mutational signatures $\left(\theta_k\right)_{k\in \mathbb{N}}\subset \Delta^{96}$ acting on the genome. Borrowing notation from the model with a finite number of mutational signatures, let $z_{ji}\in \mathbb{N}$ denote the mutational signature that generated the mutation $x_{ji}\in \{1,\ldots,96\}$. By fitting a HDP to our data we will obtain an estimate $\hat{z}_{ji}$ of $z_{ji}$. Our estimate $\hat{\theta}_k$ for the associated probability vector $\theta_k \in \Delta^{96}$ of mutational signature $k$ is</p>
<p>


$$\hat{\theta}_k=\frac{\sum_{l=1}^{96}\sum_{j=1}^N\sum_{i=1}^{M_j} \mathbb{I}(\hat{z}_{ji}=k)\mathbb{I}(x_{ji}=l)\delta_l}{\sum_{j=1}^N\sum_{i=1}^{M_j} \mathbb{I}(\hat{z}_{ji}=k)},$$

where $\delta_l\in \Delta^{96}$ has mass $1$ at position $l$. Further, we estimate the mutational signature activity $e_{kj}$ of mutational signature $k$ in sample $j$ by<br>



$$\hat{e}_{kj}=1/M_j\sum_{i=1}^{M_j} \mathbb{I}(\hat{z}_{ji}=k).$$
</p>
<h3 id="advantages-of-hdp-approach-compared-to-the-classical-nmf-approach-for-mutational-signature-estimation">Advantages of HDP approach compared to the classical NMF approach for mutational signature estimation</h3>
<p>The Hierarchical Dirichlet Process (HDP) approach offers several advantages over the classical Non-negative Matrix Factorization (NMF) method. It allows for the incorporation of prior knowledge and the imposition of group structures, such as the expectation that certain features should behave in coordinated ways <a href="https://doi.org/10.17863/CAM.22674" target="_blank" rel="noopener">(cite)</a>. In mutational signature analysis, two key tasks&mdash;(1) de novo discovery of mutational signatures and (2) estimation of signature activity&mdash;can be performed simultaneously using HDP. The method can match the observed mutational catalog to an existing signature library while also identifying new signatures by pseudo-counting the existing library as observational data. Additionally, HDP facilitates the direct learning of the number of signatures from the data, overcoming a common challenge in NMF, which often struggles to select the appropriate number of signatures. While some NMF variants can quantify uncertainty, HDP provides this capability more easily.</p>
<h3 id="references-1">References</h3>
<details>
<summary>
<p>Python Implementations 1</p>
</summary>
<ul>
<li><a href="https://radimrehurek.com/gensim/models/hdpmodel.html" target="_blank" rel="noopener">https://radimrehurek.com/gensim/models/hdpmodel.html</a></li>
<li>Gensim is a Python library for <em>topic modelling</em>, <em>document indexing</em> and <em>similarity retrieval</em> with large corpora. Target audience is the <em>natural language processing</em> (NLP) and <em>information retrieval</em> (IR) community.</li>
<li><a href="https://github.com/piskvorky/gensim/blob/develop/gensim/models/hdpmodel.py" target="_blank" rel="noopener">https://github.com/piskvorky/gensim/blob/develop/gensim/models/hdpmodel.py</a></li>
<li>based on <a href="https://github.com/blei-lab/online-hdp" target="_blank" rel="noopener">https://github.com/blei-lab/online-hdp</a></li>
<li><a href="https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-d259c237a840" target="_blank" rel="noopener">https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-d259c237a840</a></li>
<li><a href="https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-part-2-python-e5666db347a" target="_blank" rel="noopener">https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-part-2-python-e5666db347a</a></li>
<li><a href="https://github.com/ecoronado92/hdp" target="_blank" rel="noopener">https://github.com/ecoronado92/hdp</a></li>
<li><a href="https://github.com/morrisgreenberg/hdp-py" target="_blank" rel="noopener">https://github.com/morrisgreenberg/hdp-py</a></li>
</ul>
</details>
<details>
<summary>
<p>Python Implementations 2</p>
</summary>
<ul>
<li><a href="https://radimrehurek.com/gensim/models/hdpmodel.html" target="_blank" rel="noopener">https://radimrehurek.com/gensim/models/hdpmodel.html</a></li>
<li>Gensim is a Python library for <em>topic modelling</em>, <em>document indexing</em> and <em>similarity retrieval</em> with large corpora. Target audience is the <em>natural language processing</em> (NLP) and <em>information retrieval</em> (IR) community.</li>
<li><a href="https://github.com/piskvorky/gensim/blob/develop/gensim/models/hdpmodel.py" target="_blank" rel="noopener">https://github.com/piskvorky/gensim/blob/develop/gensim/models/hdpmodel.py</a></li>
<li>based on <a href="https://github.com/blei-lab/online-hdp" target="_blank" rel="noopener">https://github.com/blei-lab/online-hdp</a></li>
<li><a href="https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-d259c237a840" target="_blank" rel="noopener">https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-d259c237a840</a></li>
<li><a href="https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-part-2-python-e5666db347a" target="_blank" rel="noopener">https://towardsdatascience.com/dont-be-afraid-of-nonparametric-topic-models-part-2-python-e5666db347a</a></li>
<li><a href="https://github.com/ecoronado92/hdp" target="_blank" rel="noopener">https://github.com/ecoronado92/hdp</a></li>
<li><a href="https://github.com/morrisgreenberg/hdp-py" target="_blank" rel="noopener">https://github.com/morrisgreenberg/hdp-py</a></li>
<li>Simple Jupyter notebook: <a href="https://github.com/tdhopper/notes-on-dirichlet-processes/blob/master/pages/2015-07-30-sampling-from-a-hierarchical-dirichlet-process.ipynb" target="_blank" rel="noopener">https://github.com/tdhopper/notes-on-dirichlet-processes/blob/master/pages/2015-07-30-sampling-from-a-hierarchical-dirichlet-process.ipynb</a> (just for sampling?)</li>
</ul>
</details>
<details>
<summary>
<p>Recommended Material for HDP</p>
</summary>
<ul>
<li><a href="https://en.wikipedia.org/wiki/Hierarchical_Dirichlet_process" target="_blank" rel="noopener">https://en.wikipedia.org/wiki/Hierarchical_Dirichlet_process</a></li>
<li><a href="https://mlg.eng.cam.ac.uk/zoubin/tut06/ywt.pdf" target="_blank" rel="noopener">https://mlg.eng.cam.ac.uk/zoubin/tut06/ywt.pdf</a></li>
</ul>
</details>

    </div>

    





<div class="article-tags">
  
  <a class="badge badge-light" href="/tag/science/">Science</a>
  
</div>



<div class="share-box">
  <ul class="share">
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2Fmutational-signature-with-hierarchical-dirichlet-process%2F&amp;text=Mutational&#43;signature&#43;estimation&#43;with&#43;Hierarchical&#43;Dirichlet&#43;Process" target="_blank" rel="noopener" class="share-btn-twitter" aria-label="twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=http%3A%2F%2Flocalhost%3A1313%2Fpost%2Fmutational-signature-with-hierarchical-dirichlet-process%2F&amp;t=Mutational&#43;signature&#43;estimation&#43;with&#43;Hierarchical&#43;Dirichlet&#43;Process" target="_blank" rel="noopener" class="share-btn-facebook" aria-label="facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
        
      
      <li>
        <a href="mailto:?subject=Mutational%20signature%20estimation%20with%20Hierarchical%20Dirichlet%20Process&amp;body=http%3A%2F%2Flocalhost%3A1313%2Fpost%2Fmutational-signature-with-hierarchical-dirichlet-process%2F" target="_blank" rel="noopener" class="share-btn-email" aria-label="envelope">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2Fmutational-signature-with-hierarchical-dirichlet-process%2F&amp;title=Mutational&#43;signature&#43;estimation&#43;with&#43;Hierarchical&#43;Dirichlet&#43;Process" target="_blank" rel="noopener" class="share-btn-linkedin" aria-label="linkedin-in">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="whatsapp://send?text=Mutational&#43;signature&#43;estimation&#43;with&#43;Hierarchical&#43;Dirichlet&#43;Process%20http%3A%2F%2Flocalhost%3A1313%2Fpost%2Fmutational-signature-with-hierarchical-dirichlet-process%2F" target="_blank" rel="noopener" class="share-btn-whatsapp" aria-label="whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
  </ul>
</div>











  
  



  
  
  
    
  
  
  
  <div class="media author-card content-widget-hr">
    
      
      <a href="http://localhost:1313/"><img class="avatar mr-3 avatar-circle" src="/authors/admin/avatar_hu_8905dae0af585f9b.jpg" alt="Daniel Fridljand"></a>
    

    <div class="media-body">
      <h5 class="card-title"><a href="http://localhost:1313/">Daniel Fridljand</a></h5>
      <h6 class="card-subtitle">Software Consultant</h6>
      <p class="card-text">Software development with a strong academic background in mathematics.</p>
      <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
    <li>
      <a href="mailto:daniel.fridljand@gmail.com" >
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://scholar.google.com/citations?user=SIoMbdMAAAAJ" target="_blank" rel="noopener">
        <i class="fas fa-graduation-cap"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://github.com/FridljDa" target="_blank" rel="noopener">
        <i class="fab fa-github"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://www.linkedin.com/in/daniel-fridljand-8707a2208/" target="_blank" rel="noopener">
        <i class="fab fa-linkedin"></i>
      </a>
    </li>
  
</ul>

    </div>
  </div>


















  </div>
</article>
  </div>

  <div class="page-footer">
    
    
    <div class="container">
      <footer class="site-footer">

  












  
  
  
  
  


<p class="powered-by">
  
  <a href="/privacy/">Privacy Policy</a>
  
  
   &middot; 
  <a href="/terms/">Terms</a>
  
</p>












  
  <p class="powered-by copyright-license-text">
    © 2025 Daniel Fridljand
  </p>
  





  <p class="powered-by">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://hugoblox.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Hugo Blox Builder</a> — the free, <a href="https://github.com/HugoBlox/hugo-blox-builder" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>
</footer>

    </div>
    
  </div>

  


<script src="/js/vendor-bundle.min.js"></script>




  

  
  

  






  <script src="https://cdn.jsdelivr.net/npm/leaflet@1.7.1/dist/leaflet.min.js" integrity="" crossorigin="anonymous"></script>








  
  <script id="search-hit-fuse-template" type="text/x-template">
    <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
    </div>
  </script>
  
    <script src="https://cdn.jsdelivr.net/gh/krisk/Fuse@v3.2.1/dist/fuse.min.js" integrity="sha512-o38bmzBGX+hD3JHWUFCDA09btWaqrNmoJ3RXLlrysA7PP01Kgs4UlE4MhelE1v5dJR3+cxlR4qQlotsW7jKsnw==" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/gh/julmot/mark.js@8.11.1/dist/jquery.mark.min.js" integrity="sha512-mhbv5DqBMgrWL+32MmsDOt/OAvqr/cHimk6B8y/bx/xS88MVkYGPiVv2ixKVrkywF2qHplNRUvFsAHUdxZ3Krg==" crossorigin="anonymous"></script>
  












  
  
  
  
  
  
  

















<script id="page-data" type="application/json">{"use_headroom":true}</script>


  <script src="/js/wowchemy-headroom.js" type="module"></script>









  
  


<script src="/en/js/wowchemy.min.js"></script>



  <script src="/js/wowchemy-map.js" type="module"></script>




  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        
        <pre><code></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>


  <script src="/js/wowchemy-publication.js" type="module"></script>


















</body>
</html>
